cmd/src/update.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+
+	"github.com/tj/go-update"
+	"github.com/tj/go-update/progress"
+	"github.com/tj/go-update/stores/github"
+)
+
+const (
+	// PrivateFileMode grants owner to read/write a file.
+	PrivateFileMode = 0600
+)
+
+func init() {
+	usage := `
+Examples:
+
+	Update and replace src-cli to the recommended version:
+
+		$ src update
+`
+
+	flagSet := flag.NewFlagSet("update", flag.ExitOnError)
+
+	var apiFlags = api.NewFlags(flagSet)
+
+	handler := func(args []string) error {
+		currentDirectory, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		err = isDirWritable(currentDirectory)
+		if err != nil {
+			runWithElevatedPrivilege()
+			return nil
+		}
+
+		fmt.Printf("Current version: %s\n", buildTag)
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+		recommendedVersion, err := getRecommendedVersion(context.Background(), client)
+		if err != nil {
+			return err
+		}
+		if recommendedVersion == "" {
+			fmt.Println("Recommended Version: <unknown>")
+			return nil
+		}
+		if buildTag == recommendedVersion {
+			fmt.Printf("src-cli is already at the recommended version: %s\n", recommendedVersion)
+			return nil
+		}
+
+		// Retrieving latest release information.
+		updateManager := &update.Manager{
+			Command: "src",
+			Store: &github.Store{
+				Owner:   "sourcegraph",
+				Repo:    "src-cli",
+				Version: buildTag,
+			},
+		}
+		if runtime.GOOS == "windows" {
+			updateManager.Command += ".exe"
+		}
+		releases, err := updateManager.LatestReleases()
+		if err != nil {
+			return err
+		}
+		if len(releases) == 0 {
+			return fmt.Errorf("No latest src-cli release update")
+		}
+		latestRelease := releases[0]
+		if latestRelease.Version != recommendedVersion {
+			return fmt.Errorf("Mismatch of recommended version")
+		}
```

Comment:
Should this be an error? If you're on an older Sourcegraph instance and we recommend an older version of src-cli, why look at the latest release?

I think the check would need to be a bit more nuanced: if there's a newer patch version of the recommended version it's fine to go up. If there's a newer major/patch version it would probably be best to ask the user whether they want to upgrade to that.

(It looks like you intended to do this — there's an `Errorf` here — but the error message should include the versions, I think)